ruby_event_store/lib/ruby_event_store/mappers/protobuf.rb

```diff
@@ -12,14 +12,15 @@ def type
 
     def encode_with(coder)
       coder['event_id']   = event_id
-      coder['metadata']   = metadata
+      coder['metadata']   = ProtobufNestedStruct::HashMapStringValue.dump(metadata.each_with_object({}){|(k,v),h| h[k.to_s] =v })
       coder['data.proto'] = data.class.encode(data)
       coder['data.type']  = type
     end
 
     def init_with(coder)
       @event_id = coder['event_id']
-      @metadata = coder['metadata']
+      @metadata = Metadata.new
+      ProtobufNestedStruct::HashMapStringValue.load(coder['metadata']).each_with_object(metadata){|(k,v),meta| meta[k.to_sym] = v }
```

Comment:
Would it be more obvious that the `metadata` in `each_with_object(metadata)` also used the instance variable prefix `@`? It feels a bit confusing that one line above, we create the `@metadata` but then in here we don't use it yet.

I'm not sure about your coding style, so this comment might be irrelevant. Let me know what you think :relaxed:

Reply:
Mutant mutates `@metadata` to `metadata` and it still works so it complains. Minor ;)